pkg/reconciler/managed/reconciler.go

```diff
@@ -475,7 +488,10 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 	log := r.log.WithValues("request", req)
 	log.Debug("Reconciling")
 
-	ctx, cancel := context.WithTimeout(context.Background(), managedReconcileTimeout)
+	ctx, cancel := context.WithTimeout(context.Background(), r.timeout+reconcileTimeoutAddition)
+	defer cancel()
```

Comment:
Do we need to call `cancel()` twice - won't cancelling this parent cancel the child?

Reply:
No, we don't. It seems that it's sufficient to call parent's cancel to cancel everything. I had to add a `nolint` though because govet gives false positive about not calling cancel of a context, details here https://github.com/golang/go/issues/29587